fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -403,6 +434,314 @@ void flakyMerge(boolean isGrouped,
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(100));
+        Index index = complexPartitionedIndex(options);
+
+        Function<FDBRecordContext, Pair<FDBRecordStore, QueryPlanner>> schemaSetup = context ->
+                LuceneIndexTestUtils.rebuildIndexMetaData(context, path, TestRecordsTextProto.ComplexDocument.getDescriptor().getName(), index, useCascadesPlanner);
+
+        assertNotNull(index);
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 8)
+                .build();
+
+        final int countReps = 20;
+
+        long timestamp = System.currentTimeMillis();
+        Map<Tuple, Map<Tuple, Tuple>> insertedDocs = new HashMap<>();
+        AtomicBoolean firstRecordCommitted = new AtomicBoolean(false);
+
+        Thread inserter = new Thread(() -> {
+            for (int i = 0; i < countReps; i++) {
+                try (FDBRecordContext context = openContext(contextProps)) {
+                    Pair<FDBRecordStore, QueryPlanner> storeAndPlanner = schemaSetup.apply(context);
+                    FDBRecordStore recordStore = storeAndPlanner.getLeft();
+                    recordStore.getIndexDeferredMaintenanceControl().setAutoMergeDuringCommit(false);
+                    TestRecordsTextProto.ComplexDocument cd = TestRecordsTextProto.ComplexDocument.newBuilder()
+                            .setGroup(1)
+                            .setDocId(i + 1000L)
+                            .setIsSeen(true)
+                            .setText("A word about what I want to say")
+                            .setTimestamp(timestamp + i)
+                            .setHeader(TestRecordsTextProto.ComplexDocument.Header.newBuilder().setHeaderId(1000L - i))
+                            .build();
+                    try {
+                        Tuple primaryKey = recordStore.saveRecord(cd).getPrimaryKey();
+                        commit(context);
+                        insertedDocs.computeIfAbsent(Tuple.from(1), k -> new HashMap<>()).put(primaryKey, Tuple.from(timestamp + i));
+                    } catch (Exception e) {
+                        // commit failed due to conflict with other thread. Continue trying to create docs.
+                        LOGGER.debug("couldn't commit for key {}", (1000L + i));
+                    }
+                }
+                // after first record is committed, signal to merger record to start attempting to merge
+                firstRecordCommitted.compareAndSet(false, true);
+            }
+        });
+
+        Thread merger = new Thread(() -> {
+            // wait till first record is committed before attempting to merge
+            while (!firstRecordCommitted.get()) {
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException interruptedException) {
+                }
```

Comment:
I think this should mark the thread as interrupted, and return.
```suggestion
                } catch (InterruptedException interruptedException) {
                    Thread.currentThread().interrupt();
                    return;
                }
```